src/app/features/dashboard-widgets/entity-count-dashboard-widget/entity-count-dashboard/entity-count-dashboard.component.ts

```diff
@@ -160,27 +164,70 @@ export class EntityCountDashboardComponent
     const groupedByEntity =
```

Comment:
The `calculateGroupCounts` method already was super long. Could you try to create private sub-methods - maybe one to get groupCounts of enums, one for groupCounts of others, one to handle the notDefined?